cpp/merkletree/sparse_merkle_tree.h

```diff
@@ -171,29 +172,25 @@ class SparseMerkleTree {
   // or the leaf index is 0.
   //
   // @param path the path of the leaf whose inclusion proof to return.
-  std::vector<std::string> InclusionProof(const Path& path);
+  std::vector<std::string> InclusionProof(const Path& path) const;
 
   std::string Dump() const;
 
  private:
-  // WARNING WARNING WARNING
-  // 64 < 256 !
-  // WARNING WARNING WARNING
-  // TODO(alcutter): BIGNUM probably.
-  typedef uint64_t IndexType;
+  typedef cert_trans::BigNum IndexType;
```

Comment:
Yeah, but e.g. while a whole int for the sign?
Anyway, it's useful because in this implementation, the leaf nodes are rarely actually at the end of their paths, rather they live at the first unique prefix of their path, so the leaf nodes themselves need to store the remainder.  Currently they just store the whole thing though, leading to duplication.